pkg/controller/jenkins/configuration/user/seedjobs/seedjobs_test.go

```diff
@@ -219,3 +165,41 @@ func TestSeedJobs_isRecreatePodNeeded(t *testing.T) {
 		assert.True(t, got)
 	})
 }
+
+func TestCreateAgent(t *testing.T) {
+	t.Run("not fail when deployment is available", func(t *testing.T) {
+		// given
+		ctrl := gomock.NewController(t)
+		ctx := context.TODO()
+		defer ctrl.Finish()
+
+		namespace := "test-namespace"
+		agentName := "test-agent"
+		secret := "test-secret"
+
+		jenkinsClient := jenkinsclient.NewMockJenkins(ctrl)
+		fakeClient := fake.NewFakeClient()
+		err := v1alpha2.SchemeBuilder.AddToScheme(scheme.Scheme)
+		assert.NoError(t, err)
+
+		jenkinsClient.EXPECT().GetNodeSecret(agentName).Return(secret, nil)
+		jenkinsClient.EXPECT().GetAllNodes().Return([]*gojenkins.Node{}, nil)
+		jenkinsClient.EXPECT().CreateNode(agentName, 1, "The jenkins-operator generated agent", "/home/jenkins", agentName)
+
+		seedJobsClient := New(jenkinsClient, fakeClient, nil)
+
+		// when
+		err = fakeClient.Create(ctx, &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-deployment", agentName),
+				Namespace: namespace,
+			},
+		})
+
+		assert.NoError(t, err)
+
+		// then
+		err = seedJobsClient.createAgent(jenkinsClient, fakeClient, jenkinsCustomResource(), namespace, agentName)
+		assert.NoError(t, err)
+	})
+}
```

Comment:
add test which will delete agent deployment when seed job list is empty

Reply:
???

Reply:
done